### ISSUE TITLE
todo! and unimplemented! should generate rustc warnings when returning used values

### ISSUE THREAD

**Issue by bbarker** [rust-lang/rust#80413]
I tried this code:

```rust
pub fn some_num() -> i32 {
    todo!();
}
```

I expected to see this happen:

A warning noting that a panic would occur (either showing the line of the todo! or the line of the function call making use of the todo!, for instance). I suppose even fancier ergonomics could be added, so that if the return type is (), a () could could be supplied - but this is still not good IMO, since that implies an effect should happen, but isn't happening. So even in that case, I think a warning should be given.

Instead, this happened: 

Currently (nightly) `rustc` will not warn if some_num is used, e.g., from `main`: `println!("{}", some_num());` - this will compile without warning, but will panic:

```
$ ./target/debug/sfwtools
thread 'main' panicked at 'not yet implemented', src/sfwtools.rs:63:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Meta

`rustc --version --verbose`:

```
rustc 1.50.0-nightly (0edce6f4b 2020-12-24)
binary: rustc
commit-hash: 0edce6f4bbb4514482537f569f0b8ef48e71e0a0
commit-date: 2020-12-24
host: x86_64-unknown-linux-gnu
release: 1.50.0-nightly

```

**Comment by jonas-schievink** [comment]
Additional lints like this need to be proposed via the RFC process, so closing in favor of that
